Admin/AbstractAdmin.php

```diff
@@ -766,16 +766,28 @@ public function getFilterParameters()
     {
         $parameters = array();
 
+        // if filter persister was configured with `true`, use the default persister (session)
+        if ($this->filterPersister === true) {
+            $this->filterPersister = new SessionFilterPersister($this->request->getSession());
+        }
+
         // build the values array
         if ($this->hasRequest()) {
             $filters = $this->request->query->get('filter', array());
 
-            // if persisting filters, save filters to session, or pull them out of session if no new filters set
-            if ($this->persistFilters) {
-                if ($filters == array() && $this->request->query->get('filters') != 'reset') {
-                    $filters = $this->request->getSession()->get($this->getCode().'.filter.parameters', array());
+            // if filter persistence is configured
+            if ($this->filterPersister instanceof FilterPersisterInterface) {
```

Comment:
Not a big fan either, it was an honest question b/c I hadn't seen it could be boolean yet.